### PR TITLE
Add context to single page async client methods

### PR DESCRIPTION
**Pull request by srnagar**
Add context to single page async and paging async client methods.

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java [hunk 1]
```diff
@@ -360,6 +348,37 @@ public static ClientMethodMapper getInstance() {
...
+        withContextParameters.addAll(parameters);
+        withContextParameters.add(contextParam);
+
+        methods.add(builder
+            .parameters(withContextParameters) // update builder parameters to include context
+            .returnValue(returnValue)
+            .name(proxyMethodName)
+            .onlyRequiredParameters(false)
+            .type(clientMethodType)
+            .isGroupedParameterRequired(false)
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java:376]: Not sure whether need to set all these?

It seems to me that only thing need to be swapped is `withContextParameters`? All rest appears to be same as the `builder` from input parameter?

File: javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java [hunk 2]
```diff
@@ -196,6 +197,12 @@ public static ClientMethodMapper getInstance() {
                         .type(ClientMethodType.PagingAsyncSinglePage)
                         .isGroupedParameterRequired(false)
                         .build());
+                if (JavaSettings.getInstance().isContextClientMethodParameter()) {
```
Comment (jianghaolu) [inline javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java:200]: nit: `JavaSettings.getInstance()` is assigned to a local variable `settings` on line 53

File: javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java (hunk 1 above)
Comment (srnagar) [reply javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java:376]: Yes, you're right that the builder is already configured with these values from the previous client method but I didn't want to make an assumption that this method would be added only after the non-context method is added.

File: javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java (hunk 2 above)
Comment (srnagar) [reply javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java]: Updated to just use `settings`

**Pull request merged**